profiler/options.go

```diff
@@ -496,3 +497,15 @@ func WithLogStartup(enabled bool) Option {
 		cfg.logStartup = enabled
 	}
 }
+
+// WithLogger sets logger as the profiler's error printer.
+// Note that setting the logger for the profiler sets the same logger for the
+// tracer, and vice-versa.
+func WithLogger(logger ddtrace.Logger) Option {
```

Comment:
I don't know if this is a good approach. In hindsight, it was a bad choice for the tracer too. This setting is not app-specific, it is repo wide. Someone could come later and ask the same thing from AppSec or some other component that may come up. Or even worse, they could expect to be able to set different logger for different apps, which is wrong.

I suggest considering to make the logger exported instead...

Reply:
Do you mean make `internal/log.UseLogger` exported, as requested in #1331? Alternatively, I could keep this interface and refactor the profiler to do its own logging internally. Or we could refactor `internal/log` to provide a new `Logger` type with all the same `Error`, `Warn`, etc. methods, so that each package can use its own logger or fall back to a default, shared logger.